pkg/storage/mvcc.go

```diff
@@ -5891,14 +5891,60 @@ func MVCCCheckForAcquireLock(
 		return err
 	}
 	ltScanner, err := newLockTableKeyScanner(
-		ctx, reader, txn, str, maxLockConflicts, targetLockConflictBytes, BatchEvalReadCategory)
+		ctx, reader, txn.GetTxnMeta(), str, maxLockConflicts, targetLockConflictBytes, BatchEvalReadCategory)
 	if err != nil {
 		return err
 	}
 	defer ltScanner.close()
 	return ltScanner.scan(key)
 }
 
+// VerifyLock returns true if the supplied transaction holds a replicated lock
+// ...
+func VerifyLock(
+	ctx context.Context, reader Reader, txn *enginepb.TxnMeta, str lock.Strength, key roachpb.Key,
+) (bool, error) {
+	if txn == nil {
+		// Non-transactional requests cannot acquire locks that outlive their
+		// lifespan. Nothing to verify.
+		return false, errors.Errorf("txn must be non-nil to verify replicated lock")
+	}
+	ltScanner, err := newLockTableKeyScanner(
+		ctx, reader, txn, str, 0, 0, BatchEvalReadCategory)
+	if err != nil {
+		return false, err
+	}
+
+	defer ltScanner.close()
+	err = ltScanner.scan(key)
+	if err != nil {
+		if errors.HasType(err, (*kvpb.LockConflictError)(nil)) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	for _, iterStr := range strongerOrEqualStrengths(str) {
+		foundLock := ltScanner.foundOwn(iterStr)
+		if foundLock == nil {
+			// Proceed to check weaker strengths...
+			continue
+		}
+
+		if foundLock.Txn.Epoch != txn.Epoch {
+			continue // the lock belongs to a different epoch
+		}
+
+		// TODO(arul): add a comment explaining this.
+		if foundLock.Txn.Sequence <= txn.Sequence {
```

Comment:
Discussion in person:
- we need a `if enginepb.TxnSeqIsIgnored(foundLock.Txn.Sequence, txn.IgnoredSeqNums) { continue } else { return true }` 
- we don't need a `foundLock.Txn.Sequence <=> txn.Sequence` comparison